solarwinds_apm/distro.py

```diff
@@ -99,112 +86,69 @@ def _get_token_from_service_key(self):
-                )
-        else:
-            logger.debug(
-                "Tried to setdefault for OTLP logs with invalid protocol. Skipping."
+                headers_env_var, f"authorization=Bearer%20{header_token}"
             )
 
-    def _configure_metrics_export_env_defaults(
+    def _configure_logs_export_otlp_env_defaults(
+        self,
+        header_token: str,
+        otlp_protocol: str,
+    ) -> None:
+        """Configure env defaults for OTLP logs signal export by HTTP or gRPC to SWO"""
+        self._configure_otlp_env_defaults("logs", header_token, otlp_protocol)
+
+    def _configure_metrics_export_otlp_env_defaults(
         self,
         header_token: str,
         otlp_protocol: str,
     ) -> None:
         """Configure env defaults for OTLP metrics signal export by HTTP or gRPC to SWO"""
-        if otlp_protocol in _EXPORTER_BY_OTLP_PROTOCOL:
-            environ.setdefault(
-                OTEL_EXPORTER_OTLP_METRICS_PROTOCOL, otlp_protocol
-            )
-            environ.setdefault(
-                OTEL_METRICS_EXPORTER,
-                _EXPORTER_BY_OTLP_PROTOCOL[otlp_protocol],
-            )
-            environ.setdefault(
-                OTEL_EXPORTER_OTLP_METRICS_ENDPOINT,
-                f"{INTL_SWO_DEFAULT_OTLP_COLLECTOR}/v1/metrics",
-            )
-            if header_token:
-                environ.setdefault(
-                    OTEL_EXPORTER_OTLP_METRICS_HEADERS,
-                    f"authorization=Bearer%20{header_token}",
-                )
-        else:
-            logger.debug(
-                "Tried to setdefault for OTLP metrics with invalid protocol. Skipping."
-            )
+        self._configure_otlp_env_defaults(
+            "metrics", header_token, otlp_protocol
+        )
 
-    def _configure_traces_export_env_defaults(
+    def _configure_traces_export_otlp_env_defaults(
         self,
         header_token: str,
         otlp_protocol: Any = None,
     ) -> None:
         """Configure env defaults for OTLP traces signal export by APM protocol
         to SWO, else follow provided OTLP protocol (HTTP or gRPC)"""
-        if otlp_protocol in _EXPORTER_BY_OTLP_PROTOCOL:
-            environ.setdefault(
-                OTEL_EXPORTER_OTLP_TRACES_PROTOCOL, otlp_protocol
-            )
-            environ.setdefault(
-                OTEL_TRACES_EXPORTER, _EXPORTER_BY_OTLP_PROTOCOL[otlp_protocol]
-            )
-            environ.setdefault(
-                OTEL_EXPORTER_OTLP_TRACES_ENDPOINT,
-                f"{INTL_SWO_DEFAULT_OTLP_COLLECTOR}/v1/traces",
-            )
-            if header_token:
-                environ.setdefault(
-                    OTEL_EXPORTER_OTLP_TRACES_HEADERS,
-                    f"authorization=Bearer%20{header_token}",
-                )
-        else:
-            logger.debug(
-                "Called to setdefault for OTLP traces with empty or invalid protocol. Defaulting to SolarWinds exporter."
-            )
```

Comment:
Previously, unset or invalid `OTEL_EXPORTER_OTLP_PROTOCOL` would be the condition for setting default as APM-proto exporter. Now, it depends on if legacy.